src/main/java/org/radarcns/listener/managementportal/ManagementPortalClient.java

```diff
@@ -111,8 +112,12 @@ private static Duration parseDuration(String duration, Duration defaultValue) {
         return defaultValue;
     }
 
-    protected void updateToken(OAuth2AccessTokenDetails tokenDetails) {
-        this.token = tokenDetails;
+    protected synchronized void updateToken(OAuth2AccessTokenDetails tokenDetails) {
+        this.token = tokenDetails.getAccessToken();
+    }
+
+    private synchronized String getToken() {
```

Comment:
Why does this need to be synchronized? Is it for when a token is being updated by the above function ?

Reply:
exactly